platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -193,707 +111,80 @@ public class Browser {
-                selfId,
-                appMain.getSoftwareVersion(),
-                configAddressBook,
-                emergencyRecoveryManager);
-
-        try (loadedState) {
-            if (loadedState.isNotNull()) {
-                logger.info(
-                        STARTUP.getMarker(),
-                        new SavedStateLoadedPayload(
-                                loadedState.get().getRound(), loadedState.get().getConsensusTimestamp()));
-
-                return copyInitialSignedState(platformContext, loadedState.get());
-            }
-        }
-
-        final ReservedSignedState genesisState =
-                buildGenesisState(platformContext, configAddressBook, appMain.getSoftwareVersion(), appMain.newState());
+    private static void launchUnhandled(@NonNull final CommandLineArgs commandLineArgs) throws Exception {
+        StartupTime.markStartupTime();
+        Objects.requireNonNull(commandLineArgs, "localNodesToStart must not be null");
+        logger.info(STARTUP.getMarker(), "\n\n" + STARTUP_MESSAGE + "\n");
+        logger.debug(STARTUP.getMarker(), () -> new NodeStartPayload().toString());
 
-        try (genesisState) {
-            return copyInitialSignedState(platformContext, genesisState.get());
-        }
-    }
+        final PathsConfig pathsConfig = loadPathsConfig();
 
-    /**
-     * Load the signed state from the disk if it is present.
-     *
-     * @param mainClassName            the name of the app's SwirldMain class.
-     * @param swirldName               the name of the swirld to load the state for.
-     * @param selfId                   the ID of the node to load the state for.
-     * @param appVersion               the version of the app to use for emergency recovery.
-     * @param configAddressBook        the address book to use for emergency recovery.
-     * @param emergencyRecoveryManager the emergency recovery manager to use for emergency recovery.
-     * @return the signed state loaded from disk.
-     */
-    @NonNull
-    private ReservedSignedState getUnmodifiedSignedStateFromDisk(
-            @NonNull final PlatformContext platformContext,
-            @NonNull final String mainClassName,
-            @NonNull final String swirldName,
-            @NonNull final NodeId selfId,
-            @NonNull final SoftwareVersion appVersion,
-            @NonNull final AddressBook configAddressBook,
-            @NonNull final EmergencyRecoveryManager emergencyRecoveryManager) {
+        // Load config.txt file, parse application jar file name, main class name, address book, and parameters
+        final ApplicationDefinition appDefinition =
+                ApplicationDefinitionLoader.loadDefault(pathsConfig.getConfigPath());
+        // Determine which nodes to run locally
+        final List<NodeId> nodesToRun =
+                getNodesToRun(appDefinition.getConfigAddressBook(), commandLineArgs.localNodesToStart());
+        checkNodesToRun(nodesToRun);
```

Comment:
Can all that happen before the config is created?

Reply:
yes atm, we determine the nodes to run based on the config AB and CLI parameters. I'm expecting this to change in the future, but it is what it is atm

Reply:
IMO, the cleanest way to do this would be via configuration. But that is going to break some down stream users, and so we have to be careful when we make the change.